agent/azure-monitor-exporter/src/main/java/com/azure/monitor/opentelemetry/exporter/implementation/MetricDataMapper.java

```diff
@@ -137,13 +153,47 @@ public static void updateMetricPointBuilder(
         throw new IllegalArgumentException("metric data type '" + type + "' is not supported yet");
     }
 
+    pointBuilder.setValue(pointDataValue);
     pointBuilder.setName(metricData.getName());
-
     metricTelemetryBuilder.setMetricPoint(pointBuilder);
 
-    pointData
-        .getAttributes()
-        .forEach(
-            (key, value) -> metricTelemetryBuilder.addProperty(key.getKey(), value.toString()));
+    if (isPreAggregated) {
+      Long statusCode = pointData.getAttributes().get(SemanticAttributes.HTTP_STATUS_CODE);
+      boolean success = getSuccess(statusCode, captureHttpServer4xxAsError);
+      Boolean isSynthetic =
+          pointData
+              .getAttributes()
+              .get(AttributeKey.booleanKey("applicationinsights.internal.is_synthetic"));
+      if (metricData.getName().contains(".server.")) {
+        RequestExtractor requestExtractor =
+            new RequestExtractor(metricTelemetryBuilder, statusCode, success, isSynthetic);
+        requestExtractor.extract();
+      } else if (metricData.getName().contains(".client.")) {
+        String dependencyType =
+            metricData.getName().startsWith("http")
+                ? "http"
+                : pointData.getAttributes().get(SemanticAttributes.RPC_SYSTEM);
+        String target =
+            pointData
+                .getAttributes()
+                .get(AttributeKey.stringKey("applicationinsights.internal.target"));
+        DependencyExtractor dependencyExtractor =
+            new DependencyExtractor(
+                metricTelemetryBuilder, statusCode, success, dependencyType, target, isSynthetic);
+        dependencyExtractor.extract();
```

Comment:
is there any benefit to this being an "object" as opposed to a static util function?

Reply:
it behaves like a specific telemetry type extractor.. static doesn't feel that way.